aten/src/ATen/cpu/vec/vec128/vec128_bfloat16_neon.h

```diff
@@ -0,0 +1,475 @@
+#pragma once
+
+// DO NOT DEFINE STATIC DATA IN THIS HEADER!
+// See Note [Do not compile initializers with AVX]
+#include <ATen/cpu/vec/vec128/vec128_float_neon.h>
+#include <ATen/cpu/vec/vec128/vec128_reduced_precision_common_neon.h>
+#include <ATen/cpu/vec/vec_base.h>
+#include <c10/util/BFloat16.h>
+#include <c10/util/bit_cast.h>
+#include <c10/util/irange.h>
+
+namespace at::vec {
+// See Note [CPU_CAPABILITY namespace]
+inline
+namespace CPU_CAPABILITY {
+
+// Following vec128_half_neon.h, we only support aarch64.
+#if !defined(C10_MOBILE) && defined(__aarch64__)
+#ifdef __BIG_ENDIAN__
+#error "Big endian is not supported."
+#endif
+
+// Unlike the float16_t family of types, bfloat16_t is not available
+// when we're not targeting bfloat16 hardware support. (See
+// https://godbolt.org/z/orv6e94n4 ) So, we need to handle it at
+// uint16_t in that case.
+#ifndef __ARM_FEATURE_BF16
+using bfloat16x8_t = uint16x8_t;
+using bfloat16x4_t = uint16x4_t;
+using bfloat16_t = uint16_t;
+#define vsetq_lane_bf16 vsetq_lane_u16
```

Comment:
this macro-based approach seems problematic; the Mac SDK defines these macros even when not compiling with __ARM_FEATURE_BF16. I'll have to figure out something else.

Reply:
specifically the problem is leaking the macros into other translation units.